File: A. Word/A_Word.cpp

#include<bits/stdc++.h>
using namespace std;

string check(string word) {
    int numLower = 0, numUpper = 0;

    for(char c : word) {
        if(isupper(c)) {
            numUpper++;
        } else {
            numLower++;
        }
    }

    if(numUpper > numLower) {
        transform(word.begin(), word.end(), word.begin(), ::toupper);
        return word;
    }
    transform(word.begin(), word.end(), word.begin(), ::tolower);
    return word;
}

int main() {
    string word;
    cin >> word;

    cout << check(word);
}

File: A. Word/A_Word.py
def check(word):
    upper_case_count = 0
    for char in word:
        if char.isupper():
            upper_case_count += 1
    
    if upper_case_count > len(word) // 2:
        return word.upper()
    else:
        return word.lower()
    

if __name__ == "__main__":
    word = input()
    
    print(check(word))